### ISSUE TITLE
Preflight's RPM db scanner doesn't handle big-endian data (s390x)

### ISSUE THREAD

**Issue by jfrancin** [redhat-openshift-ecosystem/openshift-preflight#626]
## Bug Description
Case: https://gss--c.na167.visual.force.com/apex/Case_View?sbstr=03212057 (salesforce)
Project ID:             61df600a25a366740e81379c
PID:                    ospid-61df600a25a366740e81379c
Project Name:           NGINX Ingress Operator
Project Type:           Container Image
Project URL:            https://connect.redhat.com/projects/61df600a25a366740e81379c/overview
Reporter SSO:           ciarams87
Org ID:                 7644757
Originated Source:      Preflight

Partner is attempting to run a Preflight scan on an S390X container, and the two RPM tests fail with "bad magic number":

```
     - msg="check completed: HasModifiedFiles" ERROR="error detecting modified files: could not get rpm list: unexpected DB magic number: 1628767744" result="error detecting modified files: could not get rpm list: unexpected DB magic number: 1628767744"
     - msg="check completed: HasNoProhibitedPackages" ERROR="could not get rpm list: unexpected DB magic number: 1628767744" result="could not get rpm list: unexpected DB magic number: 1628767744"

```
The problem is that the RPM DB scanner makes the (false) assumption that "all the world's little-endian". Binary data at rest in an S390X container will be stored in big-endian format (native for IBM S360/370/390/Z systems).
Example code: 
https://github.com/knqyf263/go-rpmdb/blob/master/pkg/bdb/hash_metadata_page.go lines 21-30:
```
func ParseHashMetadataPage(data []byte) (*HashMetadataPage, error) {
	var metadata HashMetadataPage

	err := binary.Read(bytes.NewReader(data), binary.LittleEndian, &metadata)
	if err != nil {
		return nil, xerrors.Errorf("failed to unpack HashMetadataPage: %w", err)
	}

	return &metadata, metadata.validate()
```
and in https://github.com/knqyf263/go-rpmdb/blob/master/pkg/bdb/hash_page.go:
lines 23 to 32:
```
func ParseHashPage(data []byte) (*HashPage, error) {
	var hashPage HashPage

	err := binary.Read(bytes.NewReader(data), binary.LittleEndian, &hashPage)
	if err != nil {
		return nil, xerrors.Errorf("failed to unpack: %w", err)
	}

	return &hashPage, nil
}
```
lines 102 to 107:
```
	for idx := range hashIndexData {
		if (idx-HashIndexEntrySize)%keyValuePairSize == 0 {
			value := binary.LittleEndian.Uint16(hashIndexData[idx : idx+2])
			hashIndexValues = append(hashIndexValues, value)
		}
	}
```

Other files that have LittleEndian calls: 
https://github.com/knqyf263/go-rpmdb/blob/master/pkg/bdb/hash_page_entry.go
https://github.com/knqyf263/go-rpmdb/blob/a9e3110d8ee1fd3b0798a7abe8c59230bd265cd3/pkg/inet.go
https://github.com/knqyf263/go-rpmdb/blob/1369b2ee40b762e48586531810d5b2564e2c1063/pkg/ndb/ndb.go
https://github.com/knqyf263/go-rpmdb/blob/9cd0802af42c52825c450812d9122acf28855a4c/pkg/sqlite3/sqlite3.go
https://github.com/knqyf263/go-rpmdb/blob/0de7f922f589a6bc7073364d6057bae9ffbd858d/pkg/bdb/generic_page.go
https://github.com/knqyf263/go-rpmdb/blob/a9e3110d8ee1fd3b0798a7abe8c59230bd265cd3/pkg/entry.go

## Version and Command Invocation

not specified but likely all.

## Steps to Reproduce:
(How can we reproduce this?)
1) Run Preflight on an S390X container
2) Observe "Bad magic number" error.
3)

## Expected Result
(What did you expect to happen and why?)

The RPM db will be parsed.

## Actual Result
(What actually happened)

The parse fails with "bad magic number"

**Comment by vikasmulaje** [comment]
@jfrancin what preflight version was used?

**Comment by bcrochet** [comment]
We are aware of this issue. We're working on a fix.

**Comment by jsm84** [comment]
Working on getting a hotfix delivered now. I just sent a working binary to the customer named in the support case.

**Comment by jsm84** [comment]
@vikasmulaje This got fixed a few weeks ago upstream in go-rpmdb, and those changes were incorporated into the 1.3.1 release.

**Comment by jsm84** [comment]
Fixed by #726

**Comment by vikasmulaje** [comment]
@jsm84 Thanks for information. Yes its working fine with 1.3.1 in pipelineruns.
